packages/experimental/labs-workers/src/app/index.ts

```diff
@@ -0,0 +1,141 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { Hono } from 'hono';
+import { deleteCookie, getCookie, setCookie } from 'hono/cookie';
+import { html, raw } from 'hono/html';
+import { HTTPException } from 'hono/http-exception';
+import { decode, jwt, sign } from 'hono/jwt';
+
+import { log } from '@dxos/log';
+
+import { createMessage, sendEmail, templates } from '../api/email';
+import { UserManager } from '../api/users';
+import { DISCORD_INVITE_URL, type Env } from '../defs';
+
+const app = new Hono<Env>();
+
+/**
+ * Access control.
+ */
+app.use('/', (c, next) => {
+  const jwtMiddleware = jwt({
+    secret: c.env.JWT_SECRET,
+    cookie: 'access_token',
+  });
+
+  // https://hono.dev/middleware/builtin/jwt
+  return jwtMiddleware(c, next);
+});
+
+/**
+ * Serve app.
+ */
+// TODO(burdon): Serve app as static resource: https://github.com/honojs/examples/tree/main/serve-static
+app.get('/', async (c) => {
+  const token = getCookie(c, 'access_token');
+  if (!token) {
+    return c.redirect('/signup');
+  }
+
+  const { payload } = decode(token);
+  log.info('token', { payload });
+
+  // https://hono.dev/helpers/html
+  // TODO(burdon): Serve content HTML page from netlify.
+  return c.html(html`
+    <!doctype html>
+    <body>
+      <h1>Welcome to Composer!</h1>
+      <pre>${raw(JSON.stringify(payload, undefined, 2))}</pre>
+      <a href="/reset">Logout</a>
+    </body>
+  `);
+});
+
+/**
+ * Set JWT from one time link.
+ */
+app.get('/access', async (c) => {
```

Comment:
we're explicitly not supporting revocation, right?

Reply:
we could layer that in. but i don't think it's required initially.

Reply:
If we're using the validation endpoint to validate the token, any changes to the authorization will be immediately reflected there, right? Or am I misunderstanding something?